controllers/machineset.go

```diff
@@ -48,8 +46,8 @@ func (r *AppWrapperReconciler) scaleMachineSet(ctx context.Context, aw *arbv1.Ap
 						//klog.Infof("looking for key %v", k)
 						if k == "machine.openshift.io/cluster-api-machineset" {
 							//klog.Infof("looking for value %v", v)
-							machinePhase := *machine.Status.Phase
-							if v == copyOfaMachineSet.Name && !contains(existingMachinesOwned, v) && (machinePhase != "Deleting") {
+							machinePhase := &machine.Status.Phase
```

Comment:
Could that be `machinePhase := machine.Status.Phase` to avoid the double de-reference below?